src/utilities/delete.ts

```diff
@@ -1,11 +1,17 @@
 import { showToast } from '@prefecthq/prefect-design'
 import { localization } from '@/localization'
 
-export type itemType = 'Flow' | 'Deployment' | 'Flow run' | 'Work queue' | 'Block' | 'Notification' | 'Task run' | 'Concurrency Limit' | 'Worker pool'
+export type itemType = 'Flow' | 'Deployment' | 'Flow run' | 'Work queue' | 'Block' | 'Notification' | 'Task run' | 'Concurrency Limit' | 'Worker pool' | 'Worker pool queue'
 
-export async function deleteItem(id: string, endpoint: (arg: string) => void, type: itemType): Promise<void> {
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type DeleteItemEndpointArgs = any[] | any
+export type DeleteItemEndpoint = (...args: DeleteItemEndpointArgs[]) => Promise<void>
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export async function deleteItem(args: DeleteItemEndpointArgs, endpoint: DeleteItemEndpoint, type: itemType): Promise<void> {
   try {
-    await endpoint(id)
+    const endpointArgs = Array.isArray(args) ? args : [args]
+    await endpoint(...endpointArgs)
```

Comment:
This is a small change that should be backward compatible that will allow us to use `deleteItem` for async functions that take more than 1 argument

Reply:
honestly really not a fan of this abstraction even before this change.  Assuming we're gonna keep it, we can make the types a bit more sophisticated

we can accept generic `T` that extends `Action` (from vue-compositions, basically any function), then use Parameters so we only accept the actual params for the function, nothing more nothing less.

```ts
export async function deleteItem<T extends Action>(args: Parameters<T>, endpoint: T, type: ItemType): Promise<void> {
```

if we don't want to have to use an array for args when the action has only a single argument, we can use something like this

```ts
type MaybeSingleParam<T extends Action, Params = Parameters<T>> = Params extends [unknown] ? Params[0] | Params : Params
```

<img width="787" alt="image" src="https://user-images.githubusercontent.com/6098901/209887124-cae0a384-ea46-41d3-a7f1-79ed4379f124.png">

here's the whole file
```ts
import { showToast } from '@prefecthq/prefect-design'
import { Action } from '@prefecthq/vue-compositions'
import { localization } from '@/localization'
import { asArray } from '@/utilities/arrays'

export type ItemType = 'Flow' | 'Deployment' | 'Flow run' | 'Work queue' | 'Block' | 'Notification' | 'Task run' | 'Concurrency Limit' | 'Worker pool' | 'Worker pool queue'
type MaybeSingleParam<T extends Action, Params = Parameters<T>> = Params extends [unknown] ? Params[0] | Params : Params

export async function deleteItem<T extends Action>(args: MaybeSingleParam<T>, endpoint: T, type: ItemType): Promise<void> {
  try {
    const endpointArgs = asArray(args)
    await endpoint(...endpointArgs)

    showToast(localization.success.delete(type), 'success')
  } catch (error) {
    showToast(localization.error.delete(type.toLowerCase()), 'error')

    console.error(error)
  }
}
```

Reply:
we could also easily support return types by changing the return type from `Promise<void>` to `Promise<ReturnType<T>>` if you so desire